thinc/backends/cupy_ops.py

```diff
@@ -36,6 +36,10 @@ def to_numpy(self, data):
             return data.get()
 
     def gemm(self, x, y, out=None, trans1=False, trans2=False):
+        if type(x) == numpy.ndarray or type(y) == numpy.ndarray:
+            raise ValueError("Encountered a numpy array when processing with cupy. "
```

Comment:
We should probably rephrase this error, because there's nothing that keeps the user from calling `CupyOps.gemm` with a numpy array. If it happens in the internals, it's likely our fault – but if the user does it, we wouldn't consider that a bug and wouldn't want them to open an issue about it 😛

Reply:
Better?